x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/DownsampleShardIndexer.java

```diff
@@ -564,6 +564,15 @@ public XContentBuilder buildDownsampleDocument() throws IOException {
                 fieldProducer.write(builder);
             }
 
+            if (dimensions.length == 0) {
+                logger.debug("extracting dimensions from legacy tsid");
```

Comment:
I think there are two dimensions here:
 - Older index, using legacy tsid. This is controlled by IndexVersion.
 - Older transport layer, passing no dimensions. This is controlled by TransportVersion.

It seems possible to have a mix of older indexes with newer transport, as well as newer indexes with older transport. If so, can this fail in case we use a newer index (created in 8.13) with an updated tsid, through an older persistent task?